Add FTM and PWT driver pwm support for NXP frdm_ke17z and frdm_ke17z512

ankeXiao opened the pull request:
Tested `samples/basic/blinky_pwm`, `samples/basic/fade_led`, `tests/drivers/pwm/pwm_api` and `tests/drivers/pwm/pwm_loopback` samples.

ankeXiao committed:
dts: nxp: nxp_ke1xz.dtsi: add PWT and FTM drivers support

Add pwt and ftm drivers support for MKE17Z.

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

ankeXiao committed:
boards: nxp: frdm_ke17z and frdm_ke17z512: add ftm and pwt pwm support

Add ftm pinctrl configuration, and enable ftm2 to support pwm-led.

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

ankeXiao committed:
tests: drivers: pwm: add pwm configuration for mke17z

Add pwm configuration for frdm_ke17z and frdm_ke17z512 boards.
Set the alias of pwm-0 to ftm2 and test the pwm_api sample.
Update the ftm and pwt configuration to test pwm_loopback sample.

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

decsny commented on tests/drivers/pwm/pwm_loopback/src/test_pwm_loopback.c:
```diff
@@ -125,6 +126,11 @@ ZTEST_USER(pwm_loopback, test_pulse_capture_inverted)
 {
+#if defined(CONFIG_SOC_SERIES_KE1XZ)
+	/* Skipping normal-polarity period capture in PWT tests isn't supported by ke1xz */
+	TC_PRINT("skip the period capture test in normal polarity\n");
```
this should be handled by the driver, not by adding platform specific code to the test. You will notice that test_capture() will already skip if the driver returns -ENOTSUP when requesting this configuration.

ankeXiao replied:
Thanks for reminding, I took this code to the driver, for not supported will return -ENOTSUP

ankeXiao committed:
drivers: pwm: update pwm_mcux_pwt.c

Skip period and pulse width capture configurations that
specific flags are not supported on ke17z.

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

decsny commented on drivers/pwm/pwm_mcux_pwt.c:
```diff
@@ -85,6 +86,20 @@ static int mcux_pwt_configure_capture(const struct device *dev,
+	}
+
+	if ((flags & PWM_CAPTURE_TYPE_MASK) == PWM_CAPTURE_TYPE_PERIOD) {
+		if ((flags & PWM_POLARITY_MASK) == PWM_POLARITY_NORMAL) {
```
nit: instead of nesting conditionals, just use an &&

ankeXiao replied:
Thanks for reminding, I will update it.

decsny commented:
@ankeXiao you need to fix the merge conflicts

ankeXiao commented:
Fixed all merge conflicts

ankeXiao commented:
Hi all, please help approve and merge this PR, thanks!

ankeXiao commented:
@anangl, could you provide your review so it can move forwards?
